querydsl-core/src/main/java/com/mysema/query/DefaultQueryMetadata.java

```diff
@@ -385,8 +386,7 @@ public boolean equals(Object o) {
 
     @Override
     public int hashCode() {
-        addLastJoin();
-        return Objects.hashCode(flags, groupBy, having, joins, modifiers,
+        return Objects.hashCode(flags, groupBy, having, getJoins(), modifiers,
                 orderBy, params, projection, unique, where);
```

Comment:
I note that `where` is in hashCode() but not `equals(Object o)`. I would have expected to see it in neither or both.